app/assets/javascripts/govuk_publishing_components/analytics/explicit-cross-domain-links.js

```diff
@@ -1,32 +1,15 @@
-;(function (global) {
-  'use strict'
+window.GOVUK = window.GOVUK || {}
+window.GOVUK.Modules = window.GOVUK.Modules || {};
 
-  var GOVUK = global.GOVUK || {}
-  GOVUK.Modules = GOVUK.Modules || {}
-
-  GOVUK.Modules.ExplicitCrossDomainLinks = function () {
-    this.start = function ($module) {
-      this.element = $module[0]
-      this.attribute = 'href'
-      this.attributeValue = this.element.getAttribute(this.attribute)
-      this.eventType = 'click'
-      if (!this.attributeValue) {
-        this.attribute = 'action'
-        this.attributeValue = this.element.getAttribute(this.attribute)
-        this.eventType = 'submit'
-      }
-
-      this.handleEvent = this.handleEvent.bind(this)
-      this.handleCookiesAccepted = this.handleCookiesAccepted.bind(this)
-      // Listens for the 'submit' event if the element is a form, and the 'click' event if it is a link
-      this.element.addEventListener(this.eventType, this.handleEvent)
-    }
+(function (Modules) {
+  function ExplicitCrossDomainLinks ($module) {
+    this.$module = $module
 
     this.handleEvent = function (e) {
```

Comment:
I see that `handleEvent` was already inside the `start` function but I think normally we try to include the bulk of the JS outside of the constructor function, unless it's necessary. Can this be moved to its own function within the module at the same level as `this.init()`? i.e.

```
(function (Modules) {
  function ExplicitCrossDomainLinks ($module) {
    this.$module = $module
  }

  ExplicitCrossDomainLinks.prototype.init = function () {
    // ...
  }

  ExplicitCrossDomainLinks.prototype.handleEvent = function () {
    // ...
  }
```

Reply:
Question: what is the benefit of adding functions to the prototype which do not need to be public (i.e. there is no need to access them from outside this module as the purpose they serve is highly specific to this module)?

Reply:
I think there's a technical reason to not put too much into the constructor but I can't remember what it is unfortunately (I remember being called out on the same thing on one of my PRs in the past). Other than that I'd say the reasons were convention (generally our constructors are short, containing only the minimum of stuff i.e. `this.$module = $module`) and readability (`init` is the place where the code starts executing, so having that as near the top as possible is good).